File: mac/claude/claude-master-suite/skills/ag-seo-expert/tools/clean_bom_and_cache.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    if (isDirectory) {
      if (f !== 'vendor' && f !== 'node_modules' && f !== '.git') {
        walkDir(dirPath, callback);
      }
    } else {
      callback(path.join(dir, f));
    }
  });
}

const rootDir = '$HOME/Sites/satis';
let count = 0;

walkDir(rootDir, filePath => {
  if (filePath.endsWith('.php') || filePath.endsWith('.json') || filePath.endsWith('.html') || filePath.endsWith('.js') || filePath.endsWith('.css')) {
    const buf = fs.readFileSync(filePath);
    if (buf.length >= 3 && buf[0] === 0xEF && buf[1] === 0xBB && buf[2] === 0xBF) {
      const cleanBuf = buf.slice(3);
      fs.writeFileSync(filePath, cleanBuf);
      console.log(`[BOM TEMİZLENDİ]: ${filePath.replace(rootDir, '')}`);
      count++;
    }
  }
});

console.log(`Toplam ${count} dosyadaki UTF-8 BOM temizlendi.`);

// Cache klasörünü boşalt
const cacheDir = '$HOME/Sites/satis/storage/cache';
if (fs.existsSync(cacheDir)) {
  fs.readdirSync(cacheDir).forEach(f => {
    try {
      const fp = path.join(cacheDir, f);
      if (fs.statSync(fp).isFile()) fs.unlinkSync(fp);
    } catch (e) {}
  });
  console.log('PageCache (storage/cache) tamamen temizlendi.');
}

File: config/plugins/claude-master-suite/skills/ag-seo-expert/tools/python-araclar/clean_bom_and_cache.py
import os
import sys

def walk_dir(directory, callback):
    for root, dirs, files in os.walk(directory):
        # 'vendor', 'node_modules' ve '.git' dizinlerini hariç tut
        dirs[:] = [d for d in dirs if d not in ('vendor', 'node_modules', '.git')]
        for f in files:
            file_path = os.path.join(root, f)
            callback(file_path)

def main():
    root_dir = '$HOME/Sites/satis'
    count = 0
    target_extensions = ('.php', '.json', '.html', '.js', '.css')
    bom_marker = b'\xef\xbb\xbf'

    def process_file(file_path):
        nonlocal count
        if file_path.lower().endswith(target_extensions):
            try:
                with open(file_path, 'rb') as f:
                    buf = f.read()

                if buf.startswith(bom_marker):
                    clean_buf = buf[len(bom_marker):]
                    with open(file_path, 'wb') as f:
                        f.write(clean_buf)

                    rel_path = file_path.replace(root_dir, '')
                    print(f"[BOM TEMİZLENDİ]: {rel_path}")
                    count += 1
            except Exception as e:
                print(f"[HATA]: {file_path} işlenirken hata: {e}", file=sys.stderr)

    if os.path.exists(root_dir):
        walk_dir(root_dir, process_file)
        print(f"Toplam {count} dosyadaki UTF-8 BOM temizlendi.")
    else:
        print(f"[UYARI]: Kök dizin bulunamadı: {root_dir}")

    # Cache klasörünü boşalt
    cache_dir = '$HOME/Sites/satis/storage/cache'
    if os.path.exists(cache_dir):
        for f in os.listdir(cache_dir):
            fp = os.path.join(cache_dir, f)
            try:
                if os.path.isfile(fp):
                    os.unlink(fp)
            except Exception:
                pass
        print('PageCache (storage/cache) tamamen temizlendi.')

if __name__ == '__main__':
    main()
